session.go

```diff
@@ -658,6 +658,11 @@ func (s *Session) connect(host *HostInfo, errorHandler ConnErrorHandler) (*Conn,
 	return s.dial(host, s.connCfg, errorHandler)
 }
 
+type queryMetric struct {
+	attempts     int
```

Comment:
Make those `uint64` because there's no good reason why it should ever be negative.

Reply:
agreed.